### ISSUE TITLE
Feedback for Xiaomi - disable MIUI optimisations

### ISSUE THREAD

**Issue by jddcef** [urbandroid-team/dont-kill-my-app#396]
I find my launcher and my apps are not being killed anymore since I turned OFF MIUI Optimizations in the advanced Developers Options in the Settings.

**Comment by petrnalevka** [comment]
Hello @jddcef this is very interesting. Did not know there is such developer option, would you please share some screenshots? Is enabling developer options same on Xiaomi as on stock Android devices?

**Comment by jddcef** [comment]
Yeah it's been around for many years, but I didn't expect it would it make it better, because of the name. But it makes sense now that it is trying to over optimise the system by closing too many things.
Yes It's the standard way of clicking on Build setting in About a few times

**Comment by petrnalevka** [comment]
Would you please have a screenshot of the option in the developer settings? We will include it in the guides. Many thanks!

**Comment by jddcef** [comment]
![IMG_20220110_095735](https://user-images.githubusercontent.com/608676/148736224-bc861afc-459b-429a-83ca-b9ec480f6e83.jpg)
![IMG_20220110_095825](https://user-images.githubusercontent.com/608676/148736260-fb6193b8-af0c-414a-aa8a-23241cfd7fec.jpg)
![IMG_20220110_095900](https://user-images.githubusercontent.com/608676/148736278-ddf1066d-987b-42f6-91df-9cbe087f0cb2.jpg)
![IMG_20220110_095956](https://user-images.githubusercontent.com/608676/148736290-d2a3f0c5-691c-4122-90a6-f48bb64dffcf.jpg)

**Comment by petrnalevka** [comment]
Big thanks, very much appreciated!

**Comment by rl885** [comment]
Hello, would you be willing to send me the screenshots without the markings? I will add them to the guides. Thank you.

**Comment by jddcef** [comment]
![Screenshot_2022-01-12-10-26-05-984_com android settings](https://user-images.githubusercontent.com/608676/150073604-f006f6ae-620c-461d-86b2-083d9a4c00ed.jpg)
![Screenshot_2022-01-12-10-26-19-053_com android settings](https://user-images.githubusercontent.com/608676/150073612-6c6e0f6e-4558-4baa-82b1-aaa6d94c891f.jpg)
![Screenshot_2022-01-19-08-06-45-519_com android settings](https://user-images.githubusercontent.com/608676/150073686-5df672a4-1c17-4c83-838d-6478a48b07b2.jpg)

**Comment by jddcef** [comment]
So, I must say, turning off the Optimizations causes:
The inter app memory to be better, e.g. I can have more apps open.
BUT, it causes the memory of a single app to be too little, i.e. inside the app I can't switch between tabs without a reload of the tab for example.
Also, accessing file permissions was a problem.

If I turn the optimisation on, then the inter app lifetime isn't good, but the currently running in app memory is normal. And file access issues went away too.

So there are tradeoffs. 

I turned mine back on now. so turning it off and on again seem to have found a better middle ground on the system somehow.
My launcher is kept better in memory now after all that.

**Comment by janosroden** [comment]
The option is present in MIUI 11.0.3 (Redmi Note 5).

https://user-images.githubusercontent.com/6108742/150110126-941bc8b4-57d6-44dd-b6f8-1ad3df54921f.mp4

**Comment by rl885** [comment]
Hello, I have added the video as a gif - thank you for sharing it!

**Comment by herrbrixter** [comment]
Turning it off breaks some functionalities. It can't upload files. I opened my Gmail app and wanted to attach a photo as an attachment. I can browse the photos and select it but it never got attached.

**Comment by jddcef** [comment]
> Turning it off breaks some functionalities. It can't upload files. I opened my Gmail app and wanted to attach a photo as an attachment. I can browse the photos and select it but it never got attached.

Yes, it does mess up with uploads and file permissions somehow. 
I had good results by turning it off and then putting it back on. It seemed to have shaken loose some settings that improved my overall memory management.
But leaving it off is a bit too extreme and messes with the in app memory management while improving interapp performance. Dunno which is worse, but both are irriting. let's hope they add something good in future in terms of an update to fix memory management and not be so aggressive.

**Comment by hawtdayum** [comment]
"Turn on MIUI optimizations" is gone in MIUI 13 (android 12)
